src/pdf.sandbox.js

```diff
@@ -23,87 +24,114 @@ const pdfjsBuild = PDFJSDev.eval("BUNDLE_BUILD");
 const TESTING =
   typeof PDFJSDev === "undefined" || PDFJSDev.test("!PRODUCTION || TESTING");
 
+class SandboxSupport extends SandboxSupportBase {
+  exportValueToSandbox(val) {
+    // The communication with the Quickjs sandbox is based on strings
+    // So we use JSON.stringfy to serialize
+    return this.win.JSON.stringify(val);
+  }
+
+  importValueFromSandbox(val) {
+    // ...  and JSON.parse to deserialize
+    return this.win.JSON.parse(val);
+  }
+}
+
 class Sandbox {
-  constructor(module) {
-    this._evalInSandbox = module.cwrap("evalInSandbox", null, [
-      "string",
-      "int",
-    ]);
-    this._dispatchEventName = null;
+  constructor(win, module) {
+    this.support = new SandboxSupport(win, this);
+    module.externalCall = this.support.createSandboxExternals();
```

Comment:
PLEASE document this line here! I have asked multiple times for documentation of the non-obvious, and this is yet again another instance where this applies. Nobody who reads the code in this repository for the first time will have any clue about how this line connects to the code inside the sandbox because it is far from obvious.

Please add a comment from which it is obvious that this is used by https://github.com/mozilla/pdf.js.quickjs/blob/6836f77e568eba01e4a8f7f6e7bd666f73097892/src/myjs.js#L23-L36

This line can also benefit from some documentation:
https://github.com/mozilla/pdf.js.quickjs/blob/6836f77e568eba01e4a8f7f6e7bd666f73097892/src/pre.js#L21
... maybe even with a `console.error("externalCall called before being defined");` call to log an error to catch unexpected bugs.

and just wondering - without pre.js, would Emscripten eliminate "dead code" and prevent it from working? Or does it even work without the pre.js?